packages/developer-portal/src/components/apps/installations/download-installations-csv.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { FC, useEffect, useState } from 'react'
+import { AppSummaryModelPagedResult, InstallationModelPagedResult } from '@reapit/foundations-ts-definitions'
+import { GetActionNames, getActions } from '@reapit/utils-common'
+import { useReapitGet } from '@reapit/utils-react'
+import FileSaver from 'file-saver'
+import Papa from 'papaparse'
+import { reapitConnectBrowserSession } from '../../../core/connect-session'
+import { Button, elMb3 } from '@reapit/elements'
+
+export const downloadInstallationAction = (
+  installations: InstallationModelPagedResult,
+  apps: AppSummaryModelPagedResult,
+) => {
+  const installationWithAppNames = installations.data?.map((installation) => {
+    const appName = apps.data?.find((app) => app.id === installation.appId)?.name ?? ''
+
+    return {
+      ...installation,
+      appName,
+    }
+  })
+
+  const csv = Papa.unparse({
+    fields: [
+      'App Name',
+      'Client',
+      'Customer Name',
+      'Created On',
+      'Uninstalled On',
+      'Status',
+      'Installed By',
+      'Uninstalled By',
+    ],
+    data:
+      installationWithAppNames?.map(
+        ({ appName, client, customerName, created, terminatesOn, status, installedBy, uninstalledBy }) => [
+          appName,
+          client,
+          customerName,
+          created,
+          terminatesOn,
+          status,
+          installedBy,
+          uninstalledBy,
+        ],
+      ) || [],
+  })
+
+  const blob = new Blob([csv], { type: 'data:text/csv;charset=utf-8,' })
+  FileSaver.saveAs(blob, 'installs.csv')
+}
+
+export const DownloadInstallationsCSV: FC<{
+  appId: string
```

Comment:
appId, you can get this from `useAppsState` and `developerId` from `useGlobalState`